packages/apps/dashboard/server/src/modules/details/details.service.ts

```diff
@@ -148,40 +153,88 @@ export class DetailsService {
     return result;
   }
 
-  public async getLeadersByChainId(
+  public async getLeaders(
     chainId?: ChainId,
-    take?: number,
+    orderBy?: LeadersOrderBy,
+    orderDirection?: OrderDirection,
+    first?: number,
+    skip?: number,
   ): Promise<LeaderDto[]> {
-    const chainIds = !chainId
-      ? await this.networkConfig.getAvailableNetworks()
-      : [chainId];
+    const chainIds = chainId
+      ? [chainId]
+      : await this.networkConfig.getAvailableNetworks();
 
-    let allLeadersData: ILeader[] = [];
-    for (const id of chainIds) {
-      const leadersData = await OperatorUtils.getLeaders({ chainId: id });
-      allLeadersData = allLeadersData.concat(
-        leadersData.filter((leader) => leader.amountStaked > 0 && leader.role),
+    const results = await Promise.all(
+      chainIds.map(async (id) => {
+        const filter = this.createLeadersFilter(
+          id,
+          orderBy,
+          orderDirection,
+          first,
+          skip,
+        );
+        const [leaders, reputations] = await Promise.all([
+          OperatorUtils.getLeaders(filter),
+          this.fetchReputations(id),
+        ]);
+        return { leaders, reputations };
```

Comment:
Maybe we assign reputations to leaders here and operate only on `leaders` below?